tests/test_controller.py

```diff
@@ -230,80 +230,75 @@ def setup(self):
+            elif k == "online_registration":
+                eq_(flattened.get("online_registration"), str(expected.__dict__.get("online_registration")))
             else:
-                eq_(actual.get(k), expected.__dict__.get(k))
+                eq_(flattened.get(k), expected.__dict__.get(k))
+
+    def _check_keys(self, library):
+        # Helper method to check that the controller is sending the right pieces of information about a library.
+
+        expected_categories = ['uuid', 'basic_info', 'urls_and_contact', 'stages']
+        eq_(set(expected_categories), set(library.keys()))
+
+        expected_info_keys = ['name', 'short_name', 'description', 'timestamp', 'internal_urn', 'online_registration']
+        eq_(set(expected_info_keys), set(library.get("basic_info").keys()))
+
+        expected_url_contact_keys = ['contact_email', 'web_url', 'authentication_url', 'opds_url']
+        eq_(set(expected_url_contact_keys), set(library.get("urls_and_contact")))
+
+        expected_stage_keys = ['library_stage', 'registry_stage']
+        eq_(set(expected_stage_keys), set(library.get("stages").keys()))
+
 
     def test_libraries(self):
         # Test that the controller returns a specific set of information for each library.
+        ct = self.connecticut_state_library
+        ks = self.kansas_state_library
+        nypl = self.nypl
+
         response = self.controller.libraries()
         libraries = response.get("libraries")
 
-        expected_keys = [
-                            'uuid',
-                            'library_stage',
-                            'online_registration',
-                            'description',
-                            'short_name',
-                            'timestamp',
-                            'internal_urn',
-                            'web_url',
-                            'authentication_url',
-                            'opds_url',
-                            'registry_stage',
-                            'name',
-                            'contact_email'
-                        ]
-
         eq_(len(libraries), 3)
         for library in libraries:
-            eq_(set(library.keys()), set(expected_keys))
+            self._check_keys(library)
 
-        expected_names = [library.name for library in [self.connecticut_state_library, self.kansas_state_library, self.nypl]]
-        actual_names = [library.get("name") for library in libraries]
+        expected_names = [library.name for library in [ct, ks, nypl]]
+        actual_names = [library.get("basic_info").get("name") for library in libraries]
         eq_(set(expected_names), set(actual_names))
 
-        self._is_library(self.connecticut_state_library, libraries[0])
-        self._is_library(self.kansas_state_library, libraries[1])
-        self._is_library(self.nypl, libraries[2])
+        for idx, expected in enumerate([ct, ks, nypl]):
+            self._is_library(expected, libraries[idx])
```

Comment:
I think this would be easier to understand:

```
self._is_library(expected[0], ct)
self._is_library(expected[1], ks)
self._is_library(expected[2], nypl)
```